### DISCUSSION TITLE
What is the purpose of `serialized_params` ?

### DISCUSSION THREAD

**Discussion by marckohlbrugge** [bensheldon/good_job#1238] [Q&A] [answered]
I noticed the `serialized_params` column in the `good_jobs` table contains many of the same values as the other columns (`job_id`, `queue_name`, etc).

This made me wonder: what's the purpose of `serialized_params` ?

I was hoping to retroactively change the queue of a bunch of jobs that are already enqueued. But seeing the `queue_as` value duplicated across the record, made me think twice as updating the JSONB value seems more tricky.

**Comment by bensheldon** [comment, marked as answer]
The column `serialized_params` in the database is the serialization of a job that Active Job produces. When enqueuing a job, Active Job expects the job to be serialized out to that hash to be stored in the job backend (in this case, GoodJob), and then when executing the job, the backend passes that hash back into Active Job which deserializes/hydrates it back into a job object so Active Job can perform it.

GoodJob itself uses some of those values during queue operations or to make querying via the Dashboard easier (it's not easy or performant to reach into that serialized data). You're correct that it is duplicated.

It's totally fine to change the GoodJob columnar values directly, because that is what GoodJob itself uses for queue operations (e.g. changing queue name or priority or scheduled_at). 

I guess when the job is performed the values Active Job has will not match the values used by GoodJob e.g. if you accessed `queue_name` in your job, or retried the job, it might not reflect what GoodJob previously used if you updated the GoodJob record manually. Which maybe is something to fix, though I imagine the impact would be low. A few values are already injected into the serialized data, though nothing (yet) is overwritten:

https://github.com/bensheldon/good_job/blob/131e0cc8cfca183b736d5bdab1e348a8801cea21/app/models/good_job/base_execution.rb#L114-L121
